File: 추석_트래픽.py
def solution(lines):
    answer = 1
    N = len(lines)
    new_lines = []
    for line in lines:
        line = line.split(' ')
        e = int(line[1].split(':')[0]) * 3600 + int(line[1].split(':')[1]) * 60 + float(line[1].split(':')[2])
        s = e - float(line[2][:-1])
        s = int(s * 1000 + 1) / 1000
        new_lines.append([s,e])

    for i in range(0, N-1):
        cnt = 1
        s, e = new_lines[i]
        for j in range(i+1, N):
            if new_lines[j][0] < e+1:
                s = new_lines[j][0]
                cnt += 1
        answer = max(answer, cnt)
    return answer

File: 추석_트래픽.js
function solution(lines) {
    lines = lines.map(line => {
        line = line.split(' ');
        const S = line.slice(0, line.length - 1).join(' ');
        const T = line[line.length - 1];
        const date = new Date(S)
        const EndWithSeconds = date.getHours() * 3600 + date.getMinutes() * 60 + date.getSeconds() + date.getMilliseconds() / 1000;
        const startWithSeconds = EndWithSeconds - Number(T.substring(0, T.length - 1)) + 0.001;
        return [startWithSeconds, EndWithSeconds]
    })

    let answer = 0;
    for (let i = 0; i < lines.length; i++) {
        let tempAns = 1;
        const [s1, e1] = lines[i];
        for (let j = i + 1; j < lines.length; j++) {
            const [s2, e2] = lines[j];
            if (s2 < e1 + 1) tempAns += 1;
        }
        answer = Math.max(answer, tempAns);
    }
    return answer;
}


lines = ["2016-09-15 20:59:57.421 0.351s", "2016-09-15 20:59:58.233 1.181s", "2016-09-15 20:59:58.299 0.8s", "2016-09-15 20:59:58.688 1.041s", "2016-09-15 20:59:59.591 1.412s", "2016-09-15 21:00:00.464 1.466s", "2016-09-15 21:00:00.741 1.581s", "2016-09-15 21:00:00.748 2.31s", "2016-09-15 21:00:00.966 0.381s", "2016-09-15 21:00:02.066 2.62s"]

console.log(solution(lines))